package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,537 @@
+            }
+    }
+    _Timestep = Timestep
+    _data_keywords = ('step', 'lambda')
+
+    def __init__(self, filename, convert_units=True, **kwargs):
+        """
+        Parameters
+        ----------
+        filename : str or :class:`h5py.File`
+            trajectory filename or open h5py file
+        convert_units : bool (optional)
+            convert units to MDAnalysis units
+        **kwargs : dict
+            General reader arguments.
+
+        Raises
+        ------
+        RuntimeError
+            when `H5PY`_ is not installed
+        RuntimeError
+            when a unit is not recognized by MDAnalysis
+        ValueError
+            when ``n_atoms`` changes values between timesteps
+        ValueError
+            when ``convert_units=True`` but the H5MD file contains no units
+        NoDataError
+            when the H5MD file has no 'position', 'velocity', or
+            'force' group
+
+        """
+        if not HAS_H5PY:
+            raise RuntimeError("Please install h5py")
+        super(H5MDReader, self).__init__(filename, **kwargs)
+        self.filename = filename
+        self.convert_units = convert_units
+        self.driver = kwargs.get('driver', None)
+        self.comm = kwargs.get('comm', None)
+        self.parallel = self.driver and self.comm
+        self.open_trajectory()
+
+        # _has dictionary used for checking whether h5md file has
+        # 'position', 'velocity', or 'force' groups in the file
+        self._has = {name: name in self._particle_group for
+                     name in ('position', 'velocity', 'force')}
+
+        # Gets n_atoms from first available group
+        for name, value in self._has.items():
+            if value:
+                self.n_atoms = self._particle_group[name]['value'].shape[1]
+                break
+        else:
+            raise NoDataError("Provide at least a position, velocity"
+                              " or force group in the h5md file.")
+        self.ts = self._Timestep(self.n_atoms,
+                                 velocities=self.has_velocities,
+                                 forces=self.has_forces,
+                                 **self._ts_kwargs)
+        self.units = {'time': None,
+                      'length': None,
+                      'velocity': None,
+                      'force': None}
+        self._set_translated_units()  # fills units dictionary
+        self._read_next_timestep()
+
+    @staticmethod
+    def _format_hint(thing):
+        """Can this Reader read *thing*"""
+        # nb, filename strings can still get passed through if
+        # format='H5MD' is used
+        return HAS_H5PY and isinstance(thing, h5py.File)
+
+    def open_trajectory(self):
+        """opens the trajectory file using h5py library"""
+        self._frame = -1
+        if isinstance(self.filename, h5py.File):
+            self._file = self.filename
+        else:
+            if self.parallel:
```

Comment:
Check if it makes sense to supply only one of `comm` or `driver` — if so, we should not force the user to supply both.

Playing around with mpi4py should help.